crmsh/utils.py

```diff
@@ -1245,7 +1252,14 @@ def page_gen(g):
     w, h = get_winsize()
     if not config.core.pager or not can_ask() or options.batch:
         for line in g:
-            sys.stdout.write(term_render(line))
+            try:
+                sys.stdout.write(term_render(line))
+            except UnicodeEncodeError:
+                import io
+                correct_stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')
+                correct_stdout.write(term_render(line))
+                correct_stdout.detach()
+                del correct_stdout
```

Comment:
When using history subcommand,  due to some special characters like "->" in journal.log, sys.stdout.write will raise UnicodeEncodeError.
If just like #371 did, it also cause "ValueError: I/O operation on closed file."
So according to https://bugs.python.org/issue21363,
I "detach" it, then it works